src/components/slider/GradientSlider.tsx

```diff
@@ -1,65 +1,95 @@
 import asSliderGroupChild from './context/asSliderGroupChild';
 
+type SliderOnValueChange = (value: string, alfa: number) => void;
 
-const GRADIENT_TYPES = {
-  DEFAULT: 'default',
-  HUE: 'hue',
-  LIGHTNESS: 'lightness',
-  SATURATION: 'saturation'
-};
-
-/**
- * @description: A Gradient Slider component
- * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/SliderScreen.js
- * @gif: https://github.com/wix/react-native-ui-lib/blob/master/demo/showcase/GradientSlider/GradientSlider.gif?raw=true
- */
-class GradientSlider extends Component {
-  static displayName = 'GradientSlider';
+export enum GradientSliderTypes {
+  DEFAULT = 'default',
+  HUE = 'hue',
+  LIGHTNESS = 'lightness',
+  SATURATION = 'saturation'
+}
 
-  static propTypes = {
-    /**
+export type GradientSliderProps = {
+  /**
      * The gradient color
      */
-    color: PropTypes.string,
-    /**
+  color?: string;
+  /**
      * The gradient type (default, hue, lightness, saturation)
      */
-    type: PropTypes.oneOf(Object.values(GRADIENT_TYPES)),
-    /**
+  type?: GradientSliderTypes;
+  /**
      * The gradient steps
      */
-    gradientSteps: PropTypes.number,
-    /**
+  gradientSteps?: number;
+  /**
      * Callback for onValueChange, returns the updated color
      */
-    onValueChange: PropTypes.func
-  }
+  onValueChange?: SliderOnValueChange;
+  /**
+     * If true the component will have accessibility features enabled
+     */
+   accessible?: boolean;
+   /**
+     * The container style
+     */
+  containerStyle?: StyleProp<ViewStyle>;
+  /**
+     * If true the Slider will be disabled and will appear in disabled color
+     */
+   disabled?: boolean;
+}
 
-  static defaultProps = {
-    type: GRADIENT_TYPES.DEFAULT,
-    gradientSteps: 120
-  }
+type GradientSliderComponentProps = {
+  /**
+     * Context of the slider group
+     */
+   sliderContext: SliderContextProps;
+} & GradientSliderProps & typeof defaultProps;
+
+
+interface GradientSliderState {
+  color: tinycolor.ColorFormats.HSLA;
+  prevColor: string | undefined;
+}
+
+const defaultProps = {
+  type: GradientSliderTypes.DEFAULT,
+  gradientSteps: 120,
+  color: Colors.blue30
+};
+
+/**
+ * @description: A Gradient Slider component
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/SliderScreen.js
+ * @gif: https://github.com/wix/react-native-ui-lib/blob/master/demo/showcase/GradientSlider/GradientSlider.gif?raw=true
+ */
+class GradientSlider extends Component<GradientSliderComponentProps, GradientSliderState> {
+  static displayName = 'GradientSlider';
 
-  static types = GRADIENT_TYPES;
+  static defaultProps = defaultProps;
 
-  constructor(props) {
+  static types = GradientSliderTypes;
+
+  constructor(props: GradientSliderComponentProps) {
     super(props);
 
     this.state = {
       prevColor: props.color,
-      color: props.color ? Colors.getHSL(props.color) : undefined
```

Comment:
Leaving `this.state.color` undefined causes this component crash.  Added color in `defaultProps` to prevent this.